diff --git a/node_modules/vue3-plotly/dist/library.mjs b/node_modules/vue3-plotly/dist/library.mjs
index 572c098..06044ad 100644
--- a/node_modules/vue3-plotly/dist/library.mjs
+++ b/node_modules/vue3-plotly/dist/library.mjs
@@ -13,12 +13,13 @@ var script = {
     };
   },
 
-  props: ['data', 'layout', 'config'],
+  props: ['data', 'layout', 'config','obj'],
 
   watch: {
-    data() { this.setGraph(); },
-    layout() { this.setGraph(); },
-    config() { this.setGraph(); },
+    data() { this.resetGraph(); },
+    layout() { this.resetGraph(); },
+    config() { this.resetGraph(); },
+    obj() {this.resetGraph(); },
   },
 
   mounted() {
@@ -34,8 +35,21 @@ var script = {
 
   methods: {
     setGraph() {
-      Plotly.newPlot(this.plotlyId, this.data, this.layout, this.config);
+      if(this.obj){
+        Plotly.newPlot(this.plotlyId,this.obj);
+      }
+      else{
+        Plotly.newPlot(this.plotlyId, this.data, this.layout, this.config);
+      }
     },
+    resetGraph(){
+      if(this.obj){
+        Plotly.react(this.plotlyId,this.obj);
+      }
+      else{
+        Plotly.react(this.plotlyId, this.data, this.layout, this.config);
+      }
+    }
   },
 };
 
